### PR TITLE
Improve tenant deletion (including some related resources)

### PR DESCRIPTION
**Pull request by xxbillouxx**
Deletion is triggered by the auth_tenant_deleted event and by cron (synd_db.py)

Depends-On: https://github.com/wazo-platform/xivo-dao/pull/225
Depends-On: https://github.com/wazo-platform/xivo-dao/pull/226
Depends-On: https://github.com/wazo-platform/xivo-dao/pull/227

### CODE REVIEW

File: wazo_confd/plugins/event_handlers/plugin.py [hunk 1]
```diff
@@ -38,47 +32,16 @@ def _auth_tenant_added(self, event):
...
-
-    def _delete_config(self, config):
-        try:
-            self.provd.configs.delete(config)
-        except ProvdError as e:
-            if e.status_code != 404:
-                raise
-        except Exception:
-            pass  # device has no config
+        remove_tenant(event['uuid'])
```
Comment (pc-m) [inline wazo_confd/plugins/event_handlers/plugin.py:35]: Where does the provd devices deletion happen now?

File: wazo_confd/sync_db.py [hunk 2]
```diff
@@ -102,10 +102,11 @@ def main():
...
-        user_dao.delete(user)
+    logger.debug('Removing tenant: %s', tenant_uuid)
+    with session_scope():
+        tenant = tenant_resources_dao.get(tenant_uuid)
+        tenant_resources_dao.delete(tenant)
 
-    # FIXME(fblackburn):
-    # * Add all other resources related to tenant_uuid
-    # * Reset device to autoprov
+
```
Comment (pc-m) [inline wazo_confd/sync_db.py:110]: This is not required, the `console_scripts` in setup.py should automatically add a launcher

File: wazo_confd/plugins/event_handlers/plugin.py (hunk 1 above)
Comment (xxbillouxx) [reply wazo_confd/plugins/event_handlers/plugin.py:35]: Devices deletion in provd didn't work: the tenant is deleted, so the request to provd is denied (invalid token)

File: wazo_confd/sync_db.py (hunk 2 above)
Comment (xxbillouxx) [reply wazo_confd/sync_db.py:110]: perfect, but is it a problem if I let that ? (to be able call manually the python script)

File: wazo_confd/plugins/event_handlers/plugin.py (hunk 1 above)
Comment (pc-m) [reply wazo_confd/plugins/event_handlers/plugin.py:35]: Makes sense

**Review by pc-m — APPROVED** [review]

**Pull request merged**